internal/core/application/operator_service.go

```diff
@@ -1680,17 +1681,24 @@ func (o *operatorService) GetMarketReport(
 						volumeQuoteAmount = swapRequest.GetAmountR() + v.QuoteVolume
 					}
 
-					groupedVolume[i] = MarketVolume{
+					groupedVolumeTmp[i] = MarketVolume{
 						BaseVolume:  volumeBaseAmount,
 						QuoteVolume: volumeQuoteAmount,
 						StartTime:   v.StartTime,
 						EndTime:     v.EndTime,
 					}
+
+					groupedVolumeIndexes[i] = true
 				}
 			}
 		}
 	}
 
+	groupedVolume := make([]MarketVolume, 0)
+	for k, _ := range groupedVolumeIndexes {
+		groupedVolume = append(groupedVolume, groupedVolumeTmp[k])
+	}
+
```

Comment:
Is this piece of code skipping the 0-ed values?

Reply:
yes

Reply:
Is this related https://github.com/tdex-network/tdex-daemon/pull/566#issuecomment-1072573390

Reply:
I dont think so since there is a [test case](https://github.com/sekulicd/tdex-daemon/blob/cc2b9a9edc5fac90c9c0fe07f986506494b6d001/internal/core/application/operator_service_test.go#L257) that check length and test is passing, this is on application layer and in interface layer we are looping through array in [here](https://github.com/sekulicd/tdex-daemon/blob/cc2b9a9edc5fac90c9c0fe07f986506494b6d001/internal/interfaces/grpc/handler/operator.go#L1298) so i dont see issue in this moment.

Reply:
@Janaka-Steph are u sure u created more trades in different time slots(candle's) and that you passed correct timeRange/timeFrame args?